packages/timeline-state-resolver/src/conductor.ts

```diff
@@ -1522,3 +1589,43 @@ function removeParentFromState(
 	}
 	return o
 }
+
+/**
+ * If aborted, rejects as soon as possible, but lets the wraped function safely resolve or reject on its own
+ * @param func async function to wrap
+ * @param abortSignal the AbortSignal
+ * @returns Promise of the same type as `func`
+ */
+async function makeImmediatelyAbortable<T>(
+	func: (abortSignal?: AbortSignal) => Promise<T>,
+	abortSignal?: AbortSignal
+): Promise<T> {
+	const mainPromise = func(abortSignal)
+	if (!abortSignal) {
+		return mainPromise
+	}
+	let resolveAbortPromise: Function
+	const abortPromise = new Promise<void>((resolve, reject) => {
+		resolveAbortPromise = () => {
+			resolve()
+			// @ts-expect-error
```

Comment:
These need a reason for why we expect an error. I can't tell why this would produce something that isnt easily fixable.